client/components/dot-pager/index.jsx

```diff
@@ -74,13 +74,18 @@ export const DotPager = ( {
 	hasDynamicHeight = false,
 	children,
 	className,
+	onSetCurrentPage,
```

Comment:
Maybe this is a bit subtle, but I think I'd expect this prop to be called something more like `onPageSelected`.

`onSetCurrentPage` feels like it'd work hand-in-hand with a `currentPage` prop that used together allows the parent to fully control the pager. However `<DotPager>` is an uncontrolled component and handles all of it's on state.